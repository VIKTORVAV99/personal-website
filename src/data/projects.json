[
	{
		"title": "Electricity Maps",
		"id": "electricity-maps",
		"showcase": true,
		"type": "professional",
		"links": [
			{
				"title": "GitHub Repository",
				"url": "https://github.com/electricitymaps/electricitymaps-contrib"
			},
			{
				"title": "Live App",
				"url": "https://app.electricitymaps.com/"
			},
			{
				"title": "Company Website",
				"url": "https://electricitymaps.com/"
			}
		],
		"keywords": ["SPA", "React", "Python", "Electricity", "CO2", "Emissions", "Tailwind CSS"],
		"technologies": [
			{
				"name": "React",
				"description": "Underlying framework and the core of the project.",
				"url": "https://reactjs.org/",
				"preview": true
			},
			{
				"name": "Tailwind CSS",
				"description": "Utility-first CSS framework used to style the project.",
				"url": "https://tailwindcss.com/",
				"preview": true
			},
			{
				"name": "TypeScript",
				"description": "Superset of JavaScript with strong typing support used to ensure the type safety of the project.",
				"url": "https://www.typescriptlang.org/",
				"preview": true
			},
			{
				"name": "Python",
				"description": "Used to build the data gathering parts of the open source project.",
				"url": "https://www.python.org/",
				"preview": true
			}
		],
		"preview": {
			"description": "Electricity Maps provides actionable electricity data for more than 160 regions around the globe. The company mission is to help businesses and organizations more effectively reduce their carbon emissions, and enable products and services that empower end-users to be more carbon conscious.",
			"image": {
				"src": "images/projects/Electricity_Maps.png",
				"alt": "Electricity Maps Hero Image",
				"aspectRatio": "716/454",
				"otherFormats": ["webp", "avif"]
			}
		},
		"details": {
			"tagline": "The leading resource for 24/7 electricity COâ‚‚ data",
			"description": "I work as a Open Source Community Engineer at Electricity Maps. The Electricity Maps code base can be divided into 3 parts, our app, data gathering and backend. The app and data gathering together make up the open source project which I am responsible for maintaining. Since my primary duties are focused on maintaining the open source parts of our code I primarily work on reviewing Pull Requests from community members as well as others on the team and doing triage on issues. I also work on code directly such as creating and ensuring our CI pipeline is up to date and maintained not to mention the creation of new parsers so we can add new regions to our app and API. I also help out with shaping and implementing new features that span across the entire code base.",
			"images": []
		}
	},
	{
		"title": "Portfolio",
		"id": "portfolio",
		"showcase": true,
		"type": "personal",
		"links": [
			{
				"title": "GitHub Repository",
				"url": "https://github.com/viktorvav99/portfolio"
			}
		],
		"keywords": [
			"Portfolio",
			"Svelte",
			"SvelteKit",
			"Tailwind CSS",
			"TypeScript",
			"GitHub Actions",
			"GitHub Pages"
		],
		"technologies": [
			{
				"name": "Svelte",
				"url": "https://svelte.dev/",
				"description": "Underlying framework and the core of the project.",
				"preview": true
			},
			{
				"name": "SvelteKit",
				"url": "https://kit.svelte.dev/",
				"description": "Meta framework for Svelte with built in static site generator and routing.",
				"preview": true
			},
			{
				"name": "Tailwind CSS",
				"description": "Utility-first CSS framework used to style the project.",
				"url": "https://tailwindcss.com/",
				"preview": true
			},
			{
				"name": "Skeleton",
				"description": "UI toolkit based on TailwindCSS and SvelteKit that provides built in components and a theming system.",
				"url": "https://skeleton.dev/",
				"preview": true
			},
			{
				"name": "TypeScript",
				"description": "Superset of JavaScript with strong typing support used to ensure the type safety of the project.",
				"url": "https://www.typescriptlang.org/",
				"preview": true
			},
			{
				"name": "GitHub Actions",
				"description": "CI/CD pipeline used to automatically test and build the project on GitHub Pages when changes are pushed. It employs caching and parallelization to speed up the test and build process.",
				"url": "https://github.com/features/actions"
			},
			{
				"name": "GitHub Pages",
				"description": "Static site hosting service used to host the project. The project is deployed on GitHub Pages using the CI/CD pipeline when there are changes to the main branch.",
				"url": "https://pages.github.com/"
			}
		],
		"preview": {
			"description": "Online portfolio to showcase the projects I've worked on and the frameworks and tools I used on these projects. The portfolio is built with Svelte, SvelteKit, TailwindCSS, Skeleton and TypeScript."
		},
		"details": {
			"tagline": "My online portfolio to showcase some of the projects I've worked on and the frameworks and tools I used to build them.",
			"description": "The portfolio was built with Svelte, SvelteKit, TailwindCSS, Skeleton and TypeScript. And is deployed and hosted on GitHub Pages using an automated CI/CD pipeline with GitHub Actions to automatically publish the latest changes.",
			"images": []
		}
	},
	{
		"title": "Angular Demo",
		"id": "angular-demo",
		"showcase": true,
		"type": "personal",
		"links": [
			{
				"title": "GitHub Repository",
				"url": "https://github.com/VIKTORVAV99/MyWebsite-Angular"
			},
			{
				"title": "Live Demo",
				"url": "https://mywebsite-angular.web.app/"
			}
		],
		"keywords": [
			"SPA",
			"PWA",
			"Angular",
			"Angular CLI",
			"Angular Material",
			"Angular Fire",
			"Firebase",
			"GitHub Actions",
			"Firebase Hosting",
			"TypeScript"
		],
		"technologies": [
			{
				"name": "Angular",
				"description": "Underlying framework and the core of the project.",
				"url": "https://angular.io/",
				"preview": true
			},
			{
				"name": "Angular CLI",
				"description": "CLI utility for creating components, modules and services. It also has features that enabling rapid prototyping with a built in dev server as well as creating full production builds.",
				"url": "https://cli.angular.io/"
			},
			{
				"name": "Angular Material",
				"description": "Angular specific implementation of the Material Design Framework."
			},
			{
				"name": "Angular Fire",
				"description": "Angular specific implementation of the Firebase SDK. It provides a set of Angular services that can be used to interact with Firebase."
			},
			{
				"name": "Firebase",
				"description": "Backend as a service (BaaS) that provides a set of tools and services to build and run web and mobile applications. It provides a real-time database, authentication, storage, hosting, analytics, machine learning and more.",
				"url": "https://firebase.google.com/"
			},
			{
				"name": "TypeScript",
				"description": "Superset of JavaScript that adds static typing and other features to the language. It is used to write the code for the project.",
				"url": "https://www.typescriptlang.org/"
			},
			{
				"name": "GitHub Actions",
				"description": "CI/CD service that can be used to automate the build, test and deployment of software. It is used to automatically publish the latest changes to the project."
			}
		],
		"preview": {
			"description": "A simple Angular demo project to showcase the Angular framework and the Angular CLI."
		},
		"details": {
			"tagline": "Project to showcase the use of Angular and Firebase to build a progressive single page app.",
			"description": "The project was built with the use of Angular, Angular CLI, Angular Material, Angular Fire and Firebase. It is hosted and deployed on Firebase Hosting using a automated CI/CD pipeline with GitHub Actions to automatically publish the latest changes."
		}
	}
]
