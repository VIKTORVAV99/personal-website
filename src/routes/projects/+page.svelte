<script lang="ts">
	import type { PageData } from './$types';
	import Projects from '$components/Projects.svelte';

	export let data: PageData;
</script>

<svelte:head>
	<title>Viktor Andersson | Projects</title>
	<meta name="description" content="" />
	<meta name="keywords" content="" />
</svelte:head>

<h1>Projects</h1>

<Projects projects={data.projects} />
