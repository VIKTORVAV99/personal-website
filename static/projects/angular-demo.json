{
	"title": "Angular Demo",
	"showcase": true,
	"type": "personal",
	"links": [
		{
			"title": "GitHub Repository",
			"url": null
		}
	],
	"keywords": [
		"SPA",
		"PWA",
		"Angular",
		"Angular CLI",
		"Angular Material",
		"Angular Fire",
		"Firebase",
		"GitHub Actions",
		"Firebase Hosting",
		"TypeScript"
	],
	"technologies": [
		{
			"name": "Angular",
			"description": "Underlying framework and the core of the project.",
			"url": "https://angular.io/",
			"preview": true
		},
		{
			"name": "Angular CLI",
			"description": "CLI utility for creating components, modules and services. It also has features that enabling rapid prototyping with a built in dev server as well as creating full production builds.",
			"url": "https://cli.angular.io/"
		},
		{
			"name": "Angular Material",
			"description": "Angular specific implementation of the Material Design Framework.",
			"url": null
		},
		{
			"name": "Angular Fire",
			"description": "Angular specific implementation of the Firebase SDK. It provides a set of Angular services that can be used to interact with Firebase.",
			"url": null
		},
		{
			"name": "Firebase",
			"description": "Backend as a service (BaaS) that provides a set of tools and services to build and run web and mobile applications. It provides a real-time database, authentication, storage, hosting, analytics, machine learning and more.",
			"url": "https://firebase.google.com/"
		},
		{
			"name": "TypeScript",
			"description": "Superset of JavaScript that adds static typing and other features to the language. It is used to write the code for the project.",
			"url": "https://www.typescriptlang.org/"
		},
		{
			"name": "GitHub Actions",
			"description": "CI/CD service that can be used to automate the build, test and deployment of software. It is used to automatically publish the latest changes to the project.",
			"url": null
		}
	],
	"preview": {
		"description": "A simple Angular demo project to showcase the Angular framework and the Angular CLI.",
		"image": null
	},
	"details": {
		"tagline": "Project to showcase the use of Angular and Firebase to build a progressive single page app.",
		"description": "The project was built with the use of Angular, Angular CLI, Angular Material, Angular Fire and Firebase. It is hosted and deployed on Firebase Hosting using a automated CI/CD pipeline with GitHub Actions to automatically publish the latest changes."
	}
}
