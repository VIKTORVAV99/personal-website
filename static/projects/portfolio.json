{
	"title": "Portfolio",
	"showcase": true,
	"type": "personal",
	"links": [
		{
			"title": "GitHub Repository",
			"url": "https://github.com/viktorvav99/portfolio"
		}
	],
	"keywords": [
		"Portfolio",
		"Svelte",
		"SvelteKit",
		"Tailwind CSS",
		"TypeScript",
		"GitHub Actions",
		"GitHub Pages"
	],
	"technologies": [
		{
			"name": "Svelte",
			"url": "https://svelte.dev/",
      "preview": true
		},
		{
			"name": "SvelteKit",
			"url": "https://kit.svelte.dev/"
		},
		{
			"name": "Tailwind CSS",
			"url": "https://tailwindcss.com/"
		},
		{
			"name": "Skeleton",
			"url": "https://skeleton.dev/"
		},
		{
			"name": "TypeScript",
			"url": "https://www.typescriptlang.org/"
		},
		{
			"name": "GitHub Actions",
			"url": "https://github.com/features/actions"
		},
		{
			"name": "GitHub Pages",
			"url": "https://pages.github.com/"
		}
	],
	"preview": {
		"description": "Online portfolio to showcase the projects I've worked on and the frameworks and tools I used on these projects. The portfolio is built with Svelte, SvelteKit, TailwindCSS, Skeleton and TypeScript.",
		"image": null
	},
	"details": {
		"tagline": "My online portfolio to showcase some of the projects I've worked on and the frameworks and tools I used to build them.",
		"description": "The portfolio was built with Svelte, SvelteKit, TailwindCSS, Skeleton and TypeScript. And is deployed and hosted on GitHub Pages using an automated CI/CD pipeline with GitHub Actions to automatically publish the latest changes.",
		"images": []
	}
}
