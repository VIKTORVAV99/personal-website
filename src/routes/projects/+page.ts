import type { PageLoad } from './$types';
import projects from '$data/projects.json';
import type { ProjectConfig } from '$interfaces/projectConfig';

export const load = (() => {
	const json = projects as ProjectConfig[];
	return {
		projects: json
	};
}) satisfies PageLoad;
