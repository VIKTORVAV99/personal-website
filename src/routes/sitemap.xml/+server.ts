import projects from '$data/projects.json';

const DOMAIN = 'https://viktor.andersson.tech/';

export async function GET() {
	return new Response(
		`
        <?xml version="1.0" encoding="UTF-8" ?>
        <urlset
            xmlns="https://www.sitemaps.org/schemas/sitemap/0.9"
        >
          <url>
            <loc>${DOMAIN}</loc>
          </url>
          <url>
            <loc>${DOMAIN + 'projects'}</loc>
          </url>
          ${projects.map((projects) => `<url><loc>${DOMAIN + 'projects/' + projects.id}</loc></url>`)}
        </urlset>`
			.replaceAll(',', '')
			.trim(),
		{
			headers: {
				'Content-Type': 'application/xml'
			}
		}
	);
}
