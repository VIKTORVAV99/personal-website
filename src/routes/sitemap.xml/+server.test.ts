import { GET } from './+server';

import { describe, it, expect } from 'bun:test';
import projects from '$data/projects.json';

describe('sitemap.xml', () => {
	it('should return content type application/xml', async () => {
		const response = await GET();
		expect(response.headers.get('Content-Type')).toBe('application/xml');
	});

	it('should return 200 status', async () => {
		const response = await GET();
		expect(response.status).toBe(200);
	});

	// it should return xml that includes the domain
	it('should include domain', async () => {
		const response = await GET();
		expect(await response.text()).toContain('https://viktor.andersson.tech/');
	});

	// it should include the projects
	it('should include projects', async () => {
		const response = await GET();
		expect(await response.text()).toContain('https://viktor.andersson.tech/projects');
	});

	it('should match snapshot', async () => {
		const response = await GET();
		expect(await response.text()).toMatchSnapshot();
	});
});
