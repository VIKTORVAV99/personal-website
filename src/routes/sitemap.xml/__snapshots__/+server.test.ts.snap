// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`sitemap.xml should match snapshot 1`] = `"<?xml version="1.0" encoding="UTF-8" ?><urlset xmlns="https://www.sitemaps.org/schemas/sitemap/0.9"><url><loc>https://viktor.andersson.tech/</loc></url><url><loc>https://viktor.andersson.tech/projects</loc></url><url><loc>https://viktor.andersson.tech/projects/electricity-maps</loc></url><url><loc>https://viktor.andersson.tech/projects/portfolio</loc></url><url><loc>https://viktor.andersson.tech/projects/angular-demo</loc></url></urlset>"`;
